discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/Discovery.java

```diff
@@ -10,445 +10,721 @@
+    Validator.notNull(updateDocumentOptions, "updateDocumentOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.post(String.format("/v1/environments/%s/collections/%s/documents/%s",
+        updateDocumentOptions.environmentId(), updateDocumentOptions.collectionId(), updateDocumentOptions.documentId(
+            )));
+    builder.query(VERSION, versionDate);
+    if (updateDocumentOptions.configurationId() != null) {
+      builder.query("configuration_id", updateDocumentOptions.configurationId());
+    }
+
+    MultipartBody.Builder multipartBuilder = new MultipartBody.Builder();
+    multipartBuilder.setType(MultipartBody.FORM);
+    if (updateDocumentOptions.file() != null) {
+      MediaType mediaType = null;
+      if (updateDocumentOptions.fileMediaType() != null) {
+        mediaType = MediaType.parse(updateDocumentOptions.fileMediaType());
+      }
+      RequestBody body = InputStreamRequestBody.create(mediaType, updateDocumentOptions.file());
+      multipartBuilder.addFormDataPart("file", "filename", body);
+    }
+    if (updateDocumentOptions.metadata() != null) {
+      multipartBuilder.addFormDataPart("metadata", updateDocumentOptions.metadata());
+    }
+    if (updateDocumentOptions.configuration() != null) {
+      multipartBuilder.addFormDataPart("configuration", updateDocumentOptions.configuration());
+    }
+    builder.body(multipartBuilder.build());
+
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(DocumentAccepted.class));
+  }
+
+  /**
+   * Add an environment.
+   *
+   * Creates a new environment.  You can create only one environment per service instance. An attempt to create
+   * another environment will result in an error.  The size of the new environment can be controlled by specifying
+   * the `size` parameter. Use the table below to map size values to the size of the environment which is created:  |
+   * Size | Disk (GB)  | RAM (GB) | Included Standard Enrichments | Notes | | ---:  | -----------: | -----------: |
+   * --------------------------------------------: | -------- | | 0  | 2              | 1              | n/a
+   * (effectively unlimited)   | Free Plan only, no HA (single node in elastic.co)| | 1     | 48             | 2
+   * | 4,000    | | | 2     | 192            | 8              | 16,000   | | | 3     | 384            | 16
+   * | 32,000   | |  **Note:** you cannot set the size property when using the free plan.
+   *
+   * @param createEnvironmentOptions the {@link CreateEnvironmentOptions} containing the options for the call
+   * @return the {@link Environment} with the response
+   */
+  public ServiceCall<Environment> createEnvironment(CreateEnvironmentOptions createEnvironmentOptions) {
+    RequestBuilder builder = RequestBuilder.post("/v1/environments");
+    builder.query(VERSION, versionDate);
+    final JsonObject contentJson = new JsonObject();
+    if (createEnvironmentOptions.size() != null) {
```

Comment:
Agreed -- we need to check for createEnvironmentOptions != null above this.  Good catch.